### ISSUE TITLE
Support for file dragging

### ISSUE THREAD

**Issue by SrKomodo** [rust-windowing/winit#1499]
`winit` already has (albeit rudimentary) support for _dropping_ files onto a window, but no way to allow _dragging_ data from a window. As it already supports dropping, I think dragging should also be supported by `winit`. Being able to drag a file from a `winit` window is the last missing piece for my usecase of the library.

**Comment by toastmod** [comment]
Hey I would like this as a feature as well.

**Comment by maroider** [comment]
> Hey I would like this as a feature as well.

If you're interested in getting the ball rolling on this, then I'd suggest that you look into what APIs we need to use on each platform, and perhaps formulate why this functionality *must* exist in winit beyond "winit already supports dropping", as neither SDL2 nor GLFW support this (to my knowledge).

**Comment by toastmod** [comment]
> > Hey I would like this as a feature as well.
> 
> If you're interested in getting the ball rolling on this, then I'd suggest that you look into what APIs we need to use on each platform, and perhaps formulate why this functionality _must_ exist in winit beyond "winit already supports dropping", as neither SDL2 nor GLFW support this (to my knowledge).

Right, I have a small bit of knowledge on how this works in the windows api though I've never done it successfully in rust. I may try making a fork of winit and implementing it in my own time, probably when I will want to implement it in my own program (soon I hope).

**Comment by dhardy** [comment]
> winit already has (albeit rudimentary) support for dropping files onto a window,

Drop, yes, but `DroppedFile` and `HoveredFile` do not report the window-local coordinates and last time I tested cursor locations were not reported while dragging, so it is incomplete.

**Comment by toastmod** [comment]
@dhardy If it helps to know, the cursor coordinates are reported as a `CursorMoved` event directly after the `DroppedFile` event is fired. But at one point I was about to go on here and ask if the coordinates could be passed along with `DroppedFile` so I understand.

**Comment by toastmod** [comment]
Looks like the windows crate is finally figuring out COM objects so I might try to add this soon. But where in winit's source is the best place to put a function that initiates a file drag?

**Comment by JumpyLionnn** [comment]
are there any updates regarding this topic?

**Comment by MartorSkull** [comment]
Any news on this feature?

**Comment by toastmod** [comment]
This is now possible in the current state of the windows API for rust. For anyone interested, you are probably best to get the raw window handle from a winit window and use that with the windows API. You will have to manage OLE/COM properly though I believe or your winit event loop might break.
